src/builtins.ts

```diff
@@ -771,6 +772,47 @@ export function compileCall(
         }
       }
     }
+    case BuiltinSymbols.nameof: {
+      let value: string | null = null;
+      let resultType = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      if (resultType === null) {
+        compiler.currentType = compiler.program.stringInstance.type;
+        compiler.error(
+          DiagnosticCode.Operation_not_supported,
+          reportNode.typeArgumentsRange
+        );
+        return module.unreachable();
+      }
+
+      if (resultType.classReference !== null) {
```

Comment:
Checking for `is(TypeFlags.REFERENCE)` first can harden this a bit, for example if `i32`, that is not a reference, gets a `classReference` of `I32` in the future.

nit: Also seems that `let value: string` above can be moved above here, and doesn't have to be nullable / require an initializer anymore due to adding the default case below.

Reply:
Ah! That makes perfect sense. It might make the code clearer too. It would return `I32` instead of `i32` as well. Great catch.

Reply:
```ts
      if (resultType.is(TypeFlags.REFERENCE)) {
        if (resultType.classReference !== null) {
          value = resultType.classReference.name;
        } else if (resultType.signatureReference !== null) {
          value = "Function";
        } else {
          assert(false);
          value = "";
        }
      } else {
        switch (resultType.kind) {
```

Is this okay?

Reply:
Can be simplified to

```ts
if (resultType.classReference !== null) {
  value = resultType.classReference.name;
} else {
  assert(resultType.signatureReference);
  value = "Function";
}
```

Reply:
Perfect. Will change right now.

Reply:
All set.